### PR TITLE
build: adding provenance generation to ruby release-please workflow

### PR DESCRIPTION
**Pull request by rsoberano-ld**
**Requirements**

- [ ] I have added test coverage for new or changed functionality
- [ ] I have followed the repository's [pull request submission guidelines](../blob/main/CONTRIBUTING.md#submitting-pull-requests)
- [ ] I have validated my changes against all supported platform versions

**Related issues**

Provide links to any issues in this repository or elsewhere relating to this pull request.

**Describe the solution you've provided**

Using Github SLSA generator to generate build provenance for ruby-server-sdk

Implementation based off of previous SDK SLSA integrations with release-please

**Describe alternatives you've considered**

Provide a clear and concise description of any alternative solutions or features you've considered.

**Additional context**

Add any other context about the pull request here.

### CODE REVIEW

File: .github/workflows/manual-publish.yml [hunk 1]
```diff
@@ -34,3 +36,17 @@ jobs:
...
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      id-token: write
+      contents: write
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v1.7.0
+    with:
+      base64-subjects: "${{ needs.build-publish.outputs.gem-hash }}"
+      upload-assets: true
+      upload-tag-name: TBD
```
Comment (keelerm84) [inline .github/workflows/manual-publish.yml:51]: Same comments here as the python one. upload-assets might be good to tie into the `dry_run` flag. And TBD needs TBDed.

File: .github/workflows/manual-publish.yml [hunk 2]
```diff
@@ -34,3 +36,17 @@ jobs:
         uses: ./.github/actions/publish
```
Comment (keelerm84) [inline .github/workflows/manual-publish.yml:36]: I think you need to set the id of the step here to publish.

File: PROVENANCE.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+$ curl --location -O \
+  https://github.com/launchdarkly/ruby-server-sdk/releases/download/VERSION/multiple.intoto.jsonl
+
+# Run slsa-verifier to verify provenance against package artifacts 
+$ slsa-verifier verify-artifact \
+--provenance-path multiple-provenance.intoto.jsonl \
+--source-uri github.com/launchdarkly/ruby-server-sdk \
+launchdarkly-server-sdk-VERSION.gem
+
+TBD OUTPUT
```
Comment (keelerm84) [inline PROVENANCE.md:23]: More TBD (I know it is intentional, just don't want to forget)

File: .github/workflows/manual-publish.yml (hunk 2 above)
Comment (rsoberano-ld) [reply .github/workflows/manual-publish.yml:36]: Not sure I'm understanding this comment - `id: publish` is set in line 34(?)

Comment (keelerm84) [reply .github/workflows/manual-publish.yml:36]: 🤦🏼 just didn't see it since it was collapsed in the UI.

**Review by keelerm84 — APPROVED** [review]

**Pull request merged**